wdl-lint/src/rules/nonmatching_output.rs

```diff
@@ -0,0 +1,249 @@
+//! A lint rule to ensure each output is documented in `meta`.
+
+use indexmap::IndexMap;
+use wdl_ast::v1::MetadataSection;
+use wdl_ast::v1::OutputSection;
+use wdl_ast::v1::TaskDefinition;
+use wdl_ast::v1::TaskOrWorkflow;
+use wdl_ast::v1::WorkflowDefinition;
+use wdl_ast::AstNode;
+use wdl_ast::AstToken;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::ToSpan;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the non-matching output rule.
+const ID: &str = "NonmatchingOutput";
+
+/// Creates a "non-matching output" diagnostic.
+fn nonmatching_output(span: Span, name: &str, context: &TaskOrWorkflow) -> Diagnostic {
+    let (ty, item_name) = match context {
+        TaskOrWorkflow::Task(t) => ("task", t.name().as_str().to_string()),
+        TaskOrWorkflow::Workflow(w) => ("workflow", w.name().as_str().to_string()),
+    };
+
+    Diagnostic::warning(format!(
+        "output `{name}` is missing from `meta` section in {ty} `{item_name}`"
+    ))
+    .with_rule(ID)
+    .with_highlight(span)
+    .with_fix(format!(
+        "add output (`{name}`) key to `outputs` documentation in `meta` section"
```

Comment:
```suggestion
        "add a description of this output at `meta.outputs.{name}`"
```

Reply:
I'm not 100% sold that this verbiage is better, but wanted to see what you thought.

Reply:
Maybe this?

```suggestion
        "add a description of output `{name}` to documentation in `meta.outputs`"
```

Reply:
Yeah I'm good with that.